fix: do not keep capturing `this` when the directive gets destroyed

arturovt opened the pull request:
Hey! I've noticed a memory leak when the directive is destroyed before the promise resolves. The directive is not getting GC'd since its `this` is captured.

The `this` object may survive 2 GC sweeps and will be transitioned to the read-only space, which means the GC won't try to clean it until you're running out of memory.

![image](https://user-images.githubusercontent.com/7337691/120395540-d17d4000-c33d-11eb-84f8-1c88dff37334.png)

tomastrajan commented:
Hi @arturovt !

Thank you for a great contribution, definitely makes sense!

Would you mind doing the same also for the other directive so they both behave the same? 

Thank you!

arturovt commented:
Hey Tomas! I've addressed your feedback and updated another directive.

P.S. no functional changes there, just wrapped everything into subscription to unsubscribe further.
